src/codeu/chat/client/core/Controller.java

```diff
@@ -112,4 +112,105 @@ public ConversationHeader newConversation(String title, Uuid owner)  {
 
     return response;
   }
+
+  @Override
+  public int addUserInterest(String name, Uuid owner) {
+
+    int response = 0;
+
+    try (final Connection connection = source.connect()) {
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.NEW_USER_INTEREST_REQUEST);
+      Serializers.STRING.write(connection.out(), name);
+      Uuid.SERIALIZER.write(connection.out(), owner);
+
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.NEW_USER_INTEREST_RESPONSE) {
+        response = Serializers.INTEGER.read(connection.in());
+      } else {
+        LOG.error("Response from server failed.");
+      }
+    }
+      catch (Exception ex) {
+      System.out.println("ERROR: Exception during call on server. Check log for details.");
+      LOG.error(ex, "Exception during call on server.");
+    }
+
+    return response;
+  }
+
+  @Override
+  public int removeUserInterest(String name, Uuid owner) {
+
+    int response = 0;
+
+    try (final Connection connection = source.connect()) {
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.REMOVE_USER_INTEREST_REQUEST);
+      Serializers.STRING.write(connection.out(), name);
+      Uuid.SERIALIZER.write(connection.out(), owner);
+
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.REMOVE_USER_INTEREST_RESPONSE) {
+        response = Serializers.INTEGER.read(connection.in());
+      } else {
+        LOG.error("Response from server failed.");
+      }
+    }
+      catch (Exception ex) {
+      System.out.println("ERROR: Exception during call on server. Check log for details.");
+      LOG.error(ex, "Exception during call on server.");
+    }
+
+    return response;
+  }
+
+  @Override
+  public int addConversationInterest(String title, Uuid owner) {
+
+    int response = 0;
+
+    try (final Connection connection = source.connect()) {
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.NEW_CONVERSATION_INTEREST_REQUEST);
+      Serializers.STRING.write(connection.out(), title);
```

Comment:
Conversation titles and usernames actually aren't unique. In both cases, the one that was created first is the one that gets selected any time that the model searches by title/name (this is also the case when signing in as a user that has the same name as another user). Do you think that creating another version of this commands that take UUIDs as an argument would be the best solution for this? We originally planned to keep it simple so that users wouldn't have to type out long UUIDs every time that they wanted to change interests.